src/math/Color.js

```diff
@@ -62,39 +62,54 @@ THREE.Color.prototype = {
 
 	},
 
-	setHSL: function ( h, s, l ) {
+	setHSL: function() {
+		// comupute euclidian module of x % n
+		function emod ( x, n ) {
+			return x >= 0 ? ( x % n ) : ( ( n - ( -x % n ) ) % n );
+		}
 
-		// h,s,l ranges are in 0.0 - 1.0
+		function clamp01 ( v ) {
+			return Math.min( 1, Math.max( 0, v ));
+		}
 
-		if ( s === 0 ) {
+		return function ( h, s, l ) {
 
-			this.r = this.g = this.b = l;
+			// h,s,l ranges are in 0.0 - 1.0
+			h = emod( h, 1 );
+			s = clamp01( s );
+			l = clamp01( l );
 
-		} else {
+			if ( s <= 0 ) {
```

Comment:
Probably should be `if ( s === 0 )`, since it can't be negative after it has been clamped.